static/js/recent_topics_ui.js

```diff
@@ -344,31 +398,37 @@ export function update_topics_of_deleted_message_ids(message_ids) {
 export function filters_should_hide_topic(topic_data) {
     const msg = message_store.get(topic_data.last_msg_id);
     const sub = sub_store.get(msg.stream_id);
-
-    if (sub === undefined || !sub.subscribed) {
+    if ((sub === undefined || !sub.subscribed) && topic_data.type !== "private") {
         // Never try to process deactivated & unsubscribed stream msgs.
         return true;
     }
 
     if (filters.has("unread")) {
-        const unreadCount = unread.num_unread_for_topic(msg.stream_id, msg.topic);
+        let unreadCount;
+        if (msg.type === "private") {
+            unreadCount = unread.num_unread_for_person(msg.to_user_ids);
+        } else if (msg.type === "stream") {
+            unreadCount = unread.num_unread_for_topic(msg.stream_id, msg.topic);
+        }
```

Comment:
Extract this into `get_thread_unread_count_from_message` and use it in both extracted `format_pm/topic` function above and here.